### PR TITLE
Remove the old Salsa, rename `salsa-2022` crate to `salsa`

### LINKED ISSUE #493: Remove the old Salsa, rename salsa-2022 crate to salsa
Let's 

* remove the old salsa crate that is currently the root crate
* rename salsa-2022 to salsa and put it at the root
* remove the `salsa-tests` crate
* rename `salsa-2022-tests` to `salsa-tests`
* adjust CI scripts as needed
* maybe some other housekeeping I'm not thinking of just now?

### PR DESCRIPTION
**Pull request by mbrobbel**
Closes #493.

### CODE REVIEW

**Comment by nikomatsakis** [comment]
Woohoo!

File: book/netlify.sh [hunk 1]
```diff
@@ -14,33 +14,6 @@ curl -L https://github.com/Michael-F-Bryan/mdbook-linkcheck/releases/download/v$
...
-done
-
-# Return to the original branch or commit
-git checkout $original_branch
-
-# Copy _redirects to the root directory
-cp _redirects versions
+mdbook build
+mkdir versions
+mv book/html/* versions
```
Comment (nikomatsakis) [inline book/netlify.sh:19]: Only caveat here: I feel like maybe we should keep the older book for now. But I am happy to not have the complexity, maybe we can just "clone" the old book into the new book as a subdirectory for the time being or something.

Comment (nikomatsakis) [reply book/netlify.sh:19]: (Doesn't have to be done in this PR)

File: book/src/caveat.md [hunk 2]
```diff
@@ -1,5 +0,0 @@
-> ⚠️ **IN-PROGRESS VERSION OF SALSA.** ⚠️
->
-> This page describes the unreleased "Salsa 2022" version, which is a major departure from older versions of salsa. The code here works but is only available on github and from the `salsa-2022` crate.
->
-> If you are looking for the older version of salsa, simply visit [this link](https://salsa-rs.netlify.app/salsa)
```
Comment (nikomatsakis) [inline book/src/caveat.md:5]: We probably want some *variant* of this to indicate that it's not what is on crates.io

File: components/salsa-2022/src/debug.rs [hunk 3]
```diff
@@ -1,264 +0,0 @@
...
-/// Debug print-outs of tracked structs include the value of all their fields,
-/// but the reads of those fields are ignored by salsa. This avoids creating
-/// spurious dependencies from debugging code, but if you use the resulting
-/// string to influence the outputs (return value, accumulators, etc) from your
-/// query, salsa's dependency tracking will be undermined.
-///
-/// If for some reason you *want* to incorporate dependency output into
-/// your query, do not use the `debug` or `into_debug` helpers and instead
-/// invoke `fmt` manually.
-pub trait DebugWithDb<Db: ?Sized + AsSalsaDatabase> {
```
Comment (nikomatsakis) [inline components/salsa-2022/src/debug.rs:29]: I wonder why this didn't register as a rename?

File: components/salsa-macros/Cargo.toml [hunk 4]
```diff
@@ -11,7 +11,9 @@ description = "Procedural macros for the salsa crate"
 proc-macro = true
 
 [dependencies]
+eyre = "0.6.5"
```
Comment (nikomatsakis) [inline components/salsa-macros/Cargo.toml:14]: we use this? should probably look into that...

File: src/debug.rs [hunk 5]
```diff
@@ -1,66 +1,264 @@
...
-
-    /// Returns a lower bound on the durability for the given key.
-    /// This is typically the minimum durability of all values that
-    /// the query accessed, but we may return a lower durability in
-    /// some cases.
-    fn durability(&self, key: Self::Key) -> Durability;
-
-    /// Get the (current) set of the entries in the query table.
-    fn entries<C>(&self) -> C
+use std::{
```
Comment (nikomatsakis) [inline src/debug.rs:1]: Ah, I guess this is why. A file already existed with that path.

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
I'm going to go ahead and merge this because it seems like conflict hell

File: book/netlify.sh (hunk 1 above)
Comment (nikomatsakis) [reply book/netlify.sh:19]: Filed https://github.com/salsa-rs/salsa/issues/503

File: book/src/caveat.md (hunk 2 above)
Comment (nikomatsakis) [reply book/src/caveat.md:5]: Filed https://github.com/salsa-rs/salsa/issues/503

File: components/salsa-macros/Cargo.toml (hunk 4 above)
Comment (nikomatsakis) [reply components/salsa-macros/Cargo.toml:14]: Filed https://github.com/salsa-rs/salsa/issues/504

Comment (mbrobbel) [reply components/salsa-macros/Cargo.toml:14]: It's used in the `lazy-input` example, but not in the macro crate. I'll put up a PR.

**Pull request merged**